### ISSUE TITLE
arm64 runner improvements

### ISSUE THREAD

**Issue by jeremyd2019** [msys2/msys2-autobuild#101]
- [x] #93
- [x] Switch to new ARM64 Git for Windows
  - It's still beta, but would probably work ok, and we can get it some exercise. 
  - Done in jeremyd2019/winautoconfig@f7b8399376544f8cd43270b8fe96348bbb8eef25
- [ ] Switch to ARM64 Python
  - just remove the `architecture: x64`.  I tested, but there are not yet binary wheels for win-arm64, so installing requirements fails for lack of visual studio.  I'm not interested in installing visual studio just for that. 
- [ ] Update to Windows 11 24H2
  - I did some preliminary work (and random commentary) on that in https://github.com/jeremyd2019/winautoconfig/issues/5, but I haven't been offered the update yet on any system (though I didn't opt in for latest updates).  I'd like to wait until the host gets the update at least.  Based on my previous feelings on 'major' Windows upgrades, I'd normally wait a few months for the bugs to be shaken out of the new version.

**Comment by lazka** [comment]
>     * [ ]  Switch to ARM64 Python

None of the packages affected provide arm64 binary wheels. They are likely all waiting for GitHub hosted runners (for example https://github.com/pyca/cryptography/issues/11199) as Python packaging does not support cross compiling (except hackily with setuptools which is on the way out). So I don't expect much progress there. The timeline for those runners was recently moved/adjusted to Q1 2025 (https://github.com/github/roadmap/issues/970)

**Comment by jeremyd2019** [comment]
I've added [pre and post job scripts/hooks](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job) on my ARM64 runners: https://github.com/jeremyd2019/winautoconfig/compare/202e943d2bf03ca0e4f00b6a733e3e956d23ed6b...d4a11a54efb8725c7bd9b6e21b893197bbdb7add

**Comment by lazka** [comment]
> I've added [pre and post job scripts/hooks](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job) on my ARM64 runners: [jeremyd2019/winautoconfig@202e943...3419a78](https://github.com/jeremyd2019/winautoconfig/compare/202e943d2bf03ca0e4f00b6a733e3e956d23ed6b...3419a788ac86dc73ed55776f13916cde148d3be7)

Does that mean we can cancel suck jobs now?

**Comment by lazka** [comment]
I tried it, seems to have worked (or it was in a good state): https://github.com/msys2-arm/msys2-autobuild/actions/runs/12702640749/job/35409213279

**Comment by jeremyd2019** [comment]
Yes, it now should be safer to cancel jobs.  Not completely safe, but safer.  The fact that it was not building a package means it probably would have been relatively safe before.

> oh, now it's stuck again in make_tree_writable

That's https://github.com/git-for-windows/git/issues/5320

**Comment by jeremyd2019** [comment]
maybe make_tree_writable needs to detect symlinks/directory junctions and not follow them?  Just thinking out loud

**Comment by lazka** [comment]
huh, good point, https://github.com/python/cpython/issues/67596

**Comment by lazka** [comment]
I've created #107

**Comment by jeremyd2019** [comment]
I'm installing the arm64 git-for-windows on my runners with this month's windows updates.

**Comment by jeremyd2019** [comment]
Ugh, the windows 11 23h2 update KB5051989 is breaking remote desktop (at least on arm64 systems and local accounts) for me, fails with a quick flash of a device attached to the system is not functioning.  Might have to skip this one (or wait if MS comes up with a patch for the patch)

**Comment by jeremyd2019** [comment]
I ~lied~ misspoke, I updated the runner images with the February updates, but the host was downgraded to January due to not being able to use remote desktop (I don't want to have to dig out the USB-C-to-HDMI dongle and use it locally).

**Comment by jeremyd2019** [comment]
March's update has remote desktop working properly again.
